src/EnergyPlus/GroundTemperatureModeling/KusudaAchenbachGroundTemperatureModel.cc

```diff
@@ -0,0 +1,262 @@
+// C++ Headers
+#include<memory>
+
+// ObjexxFCL Headers
+
+// EnergyPlus Headers
+#include <DataIPShortCuts.hh>
+#include <GroundTemperatureModeling/KusudaAchenbachGroundTemperatureModel.hh>
+#include <GroundTemperatureModeling/GroundTemperatureModelManager.hh>
+#include <InputProcessor.hh>
+#include <WeatherManager.hh>
+
+namespace EnergyPlus {
+
+	//******************************************************************************
+
+	// Kusuda model factory
+	std::shared_ptr< KusudaGroundTempsModel > 
+	KusudaGroundTempsModel::KusudaGTMFactory( 
+		int objectType, 
+		std::string objectName
+	)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Matt Mitchell
+		//       DATE WRITTEN   Summer 2015
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Reads input and creates instance of Kusuda ground temps model
+
+		// USE STATEMENTS:
+		using DataGlobals::SecsInDay;
+		using namespace DataIPShortCuts;
+		using namespace GroundTemperatureManager;
+
+		// Locals
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:	
+		bool found = false;
+		int NumNums;
+		int NumAlphas;
+		int IOStat;
+		bool ErrorsFound = false;
+
+		// New shared pointer for this model object
+		std::shared_ptr< KusudaGroundTempsModel > thisModel( new KusudaGroundTempsModel() );
+
+		std::string const cCurrentModuleObject = CurrentModuleObjects( objectType_KusudaGroundTemp );
+		int numCurrModels = InputProcessor::GetNumObjectsFound( cCurrentModuleObject );
+
+		for ( int modelNum = 1; modelNum <= numCurrModels; ++modelNum ) {
+
+			InputProcessor::GetObjectItem( cCurrentModuleObject, modelNum, cAlphaArgs, NumAlphas, rNumericArgs, NumNums, IOStat );
+
+			if ( objectName == cAlphaArgs( 1 ) ) {
+
+				// Read input into object here
+				thisModel->objectName = cAlphaArgs( 1 );
+				thisModel->objectType = objectType;
+				thisModel->groundThermalDiffisivity = rNumericArgs( 1 ) / ( rNumericArgs( 2 ) * rNumericArgs( 3 ) );
+
+				bool useGroundTempDataForKusuda = rNumericArgs( 4 ) || rNumericArgs( 5 ) || rNumericArgs( 6 );
+
+				if ( useGroundTempDataForKusuda ) {
+					// Use Kusuda Parameters
+					thisModel->aveGroundTemp = rNumericArgs( 4 );
+					thisModel->aveGroundTempAmplitude = rNumericArgs( 5 );
+					thisModel->phaseShiftInSecs = rNumericArgs( 6 ) * SecsInDay;
+				} else {
+					// Use data from Site:GroundTemperature:Shallow to generate parameters
+
+					int monthsInYear( 12 );
+					int avgDaysInMonth( 30 );
+					int monthOfMinSurfTemp( 0 );
+					Real64 averageGroundTemp( 0 );
+					Real64 amplitudeOfGroundTemp( 0 );
+					Real64 phaseShiftOfMinGroundTempDays( 0 );
+					Real64 minSurfTemp( 100 ); // Set high month 1 temp will be lower and actually get updated
+					Real64 maxSurfTemp( -100 ); // Set low initially but will get updated
+
+					std::shared_ptr< BaseGroundTempsModel > shallowObj = GetGroundTempModelAndInit( CurrentModuleObjects( objectType_SiteShallowGroundTemp ), "" );
```

Comment:
And this is how `shared_ptr` spreads like a virus. Because an object is *used* by multiple other objects is not justification for `shared_ptr`: only if they must share ownership because the object must stay alive until the last of them is done with it (or itself destroyed) do you need/want shared ownership. I didn't look into this instance so I'm not sure the use case for the factory is returning pre-existing objects but that is unusual: normally I'd think of such behavior as occurring outside the factory and only calling the factory if you determine you "need a new one". But this situation may justify the unusual design.

Normally the pattern of a factory returning `unique_ptr` is best: it can be loaded into a `shared_ptr` be the callee as @mbadams5 shows.